packages/blast-core/lib/cudos-contract.js

```diff
@@ -101,10 +99,11 @@ module.exports.CudosContract = class CudosContract {
         'Use "instantiate" for uploaded contracts')
     }
     options.signer = options.signer ?? await getDefaultSigner()
-    const uploadTx = await this.#uploadContract(options.signer)
+    const uploadTx = await this.#uploadContract(options.signer, GAS_AUTO, GAS_AUTO)
```

Comment:
why the gas limit and multilier are hardcoded here, but in the `instantiate` are not?

Reply:
deploy() uses 2 different TXs: upload and instantiate. So I have 3 options here to pass on deploy()
1. pass the same gas limit and multiplier for both TXs which would be weird and confusing
2. pass `uploadGasLimit`, `uploadGasMultiplier`, `instantiateGasLimit`, `instantiateGasMultiplier` to deploy function. I don't like this because we want deploy() to not bother the user about upload/instantiate
3. pass hardcoded values `auto`
4. do not pass anything and let the logic handle 'undefined' values.
Users can always use `uploadCode()` and then `instantiate()` if they want to specify gas limit and multiplier

Reply:
Understood, lets go with the current option with the auto stuff